ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -517,6 +527,71 @@ public void shouldFailCreateTableWithInferenceWithIncorrectKey() throws Exceptio
     assertThat(result.get(0), instanceOf(ErrorMessageEntity.class));
   }
 
+  private void validateQueryDescription(
+      String ksqlQueryString,
+      Map<String, Object> overriddenProperties,
+      KsqlEntity entity) throws Exception {
+    assertThat(entity, instanceOf(SourceDescription.class));
+    QueryMetadata queryMetadata = ksqlEngine.buildMultipleQueries(
+        ksqlQueryString, overriddenProperties).get(0);
+    validateQueryDescription(queryMetadata, overriddenProperties, entity);
+  }
+
+  private void validateQueryDescription(
+      QueryMetadata queryMetadata,
+      Map<String, Object> overriddenProperties,
+      KsqlEntity entity) {
+    SourceDescription sourceDescription = (SourceDescription) entity;
```

Comment:
Perhaps it would be cleaner to `assertThat(entity, instanceOf(SourceDescription.class))`, and then do the cast.